### PR TITLE
feat(metrics): expose events metrics

### PR DESCRIPTION
**Pull request by aurrelhebert**

### CODE REVIEW

File: cmd/root.go [hunk 1]
```diff
@@ -60,38 +68,89 @@ func initConfig() {
 	}
 }
 
+var (
+	eventsCounter = prometheus.NewCounter(prometheus.CounterOpts{
+		Namespace: "ssl",
```
Comment (miton18) [inline cmd/root.go:72]: What about:
namespace -> ssl_agent / http_agent
SubSystem -> queue / api
name .... as this

File: cmd/root.go [hunk 2]
```diff
@@ -16,6 +19,11 @@ import (
 )
 
 func init() {
+	prometheus.MustRegister(eventsCounter)
+	prometheus.MustRegister(eventsFatalErrorCounter)
+	prometheus.MustRegister(eventsErrorCounter)
+	prometheus.MustRegister(eventsKafkaConsumerErrorCounter)
+	prometheus.MustRegister(sslHaltedCounter)
```
Comment (miton18) [inline cmd/root.go]: Can you also register `eventsSchedulerErrorCounter` ?

File: cmd/root.go [hunk 3]
```diff
@@ -60,38 +68,89 @@ func initConfig() {
 	}
 }
 
+var (
+	eventsCounter = prometheus.NewCounter(prometheus.CounterOpts{
+		Namespace: "ssl",
+		Subsystem: "agent",
+		Name:      "events",
+		Help:      "Number of events handled.",
```
Comment (miton18) [inline cmd/root.go]: ending `.` can be drop

File: cmd/root.go [hunk 4]
```diff
@@ -60,38 +68,89 @@ func initConfig() {
...
+		Subsystem: "agent",
+		Name:      "scheduler_error_events",
+		Help:      "Number of events causing scheduler error",
+	})
+
+	sslHaltedCounter = prometheus.NewCounter(prometheus.CounterOpts{
+		Namespace: "ssl",
+		Subsystem: "agent",
+		Name:      "halts",
+		Help:      "Number of ssl agent  halts",
```
Comment (miton18) [inline cmd/root.go]: Double spaces

File: cmd/root.go [hunk 5]
```diff
@@ -60,38 +68,89 @@ func initConfig() {
...
+		http.ListenAndServe(":2112", nil)
+
 		quit := make(chan os.Signal, 2)
 		signal.Notify(quit, syscall.SIGTERM)
 		signal.Notify(quit, syscall.SIGINT)
 
 		consumer, err := core.NewConsumer()
 		if err != nil {
+			eventsFatalErrorCounter.Inc()
 			log.WithError(err).Fatal("Could not start the kafka consumer")
```
Comment (miton18) [inline cmd/root.go]: if you fatal here, the metric you increment just before, won't have the time to be scrapped, you will never see this one

File: cmd/root.go [hunk 6]
```diff
@@ -60,38 +68,89 @@ func initConfig() {
...
 		}
 
 		for {
 			select {
 			case m, ok := <-consumer.Messages():
 				if !ok {
+					eventsFatalErrorCounter.Inc()
```
Comment (miton18) [inline cmd/root.go]: if you fatal here, the metric you increment just before, won't have the time to be scrapped, you will never see this one

**Review by miton18 — CHANGES_REQUESTED** [review]

File: cmd/root.go [hunk 7]
```diff
@@ -100,11 +150,14 @@ var RootCmd = &cobra.Command{
...
 
 			case err := <-consumer.Errors():
+				eventsKafkaConsumerErrorCounter.Inc()
 				log.WithError(err).Error("Kafka consumer error")
 
 			case notif := <-consumer.Notifications():
+				eventsCounter.Inc()
 				log.Info(fmt.Sprintf("%+v", notif))
 			case <-quit:
+				sslHaltedCounter.Inc()
```
Comment (FlorentinDUBOIS) [inline cmd/root.go]: It is not useful here because this metrics will never be scraped

File: cmd/root.go [hunk 8]
```diff
@@ -60,13 +68,53 @@ func initConfig() {
...
+	})
+
+	eventsSchedulerErrorCounter = prometheus.NewCounter(prometheus.CounterOpts{
+		Namespace: "ssl",
+		Subsystem: "agent",
+		Name:      "scheduler_error_events",
+		Help:      "Number of events causing scheduler error",
+	})
+
+	sslHaltedCounter = prometheus.NewCounter(prometheus.CounterOpts{
```
Comment (FlorentinDUBOIS) [inline cmd/root.go]: Can you remove this counter?

**Review by FlorentinDUBOIS — CHANGES_REQUESTED** [review]

File: cmd/root.go [hunk 9]
```diff
@@ -60,13 +67,46 @@ func initConfig() {
...
+
 // RootCmd launch the aggregator agent.
 var RootCmd = &cobra.Command{
 	Use:   "poke-ssl-agent",
 	Short: "poke-ssl-agent collect SSL domains stats",
 	Run: func(cmd *cobra.Command, args []string) {
 		log.Info("poke-ssl-agent starting")
 
+		http.Handle("/metrics", promhttp.Handler())
+		http.ListenAndServe(":2112", nil)
```
Comment (miton18) [inline cmd/root.go]: Test the error

**Review by miton18 — CHANGES_REQUESTED** [review]

**Review by miton18 — APPROVED** [review]

**Review by FlorentinDUBOIS — APPROVED** [review]

**Pull request merged**